src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -714,106 +599,84 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
     // Arguments:
     // - keys: the key chord of the command to search for
     // Return Value:
-    // - the command with the given key chord
-    // - nullptr if the key chord is explicitly unbound
-    // - nullopt if it was not bound in this layer
-    std::optional<Model::Command> ActionMap::_GetActionByKeyChordInternal(const Control::KeyChord& keys) const
+    // - the command ID with the given key chord
+    // - an empty string if the key chord is explicitly unbound
+    // - nullopt if it is not bound
+    std::optional<winrt::hstring> ActionMap::_GetActionIdByKeyChordInternal(const Control::KeyChord& keys) const
     {
-        // Check the current layer
-        if (const auto actionIDPair = _KeyMap.find(keys); actionIDPair != _KeyMap.end())
+        if (const auto keyIDPair = _KeyMap.find(keys); keyIDPair != _KeyMap.end())
         {
-            // the command was explicitly bound,
-            // return what we found (invalid commands exposed as nullptr)
-            return _GetActionByID(actionIDPair->second);
+            // the keychord is defined in this layer, return the ID (ID be empty, in which case this key is explicitly unbound)
+            return keyIDPair->second;
         }
 
-        // the command was not bound in this layer,
-        // ask my parents
+        // search through our parents
         for (const auto& parent : _parents)
         {
-            const auto& inheritedCmd{ parent->_GetActionByKeyChordInternal(keys) };
-            if (inheritedCmd)
+            if (const auto foundCmdID = parent->_GetActionIdByKeyChordInternal(keys))
             {
-                return *inheritedCmd;
+                return foundCmdID;
             }
         }
 
-        // This action is not explicitly bound
+        // we did not find the keychord anywhere, its not bound and not explicity unbound either
         return std::nullopt;
     }
 
     // Method Description:
-    // - Retrieves the key chord for the provided action
+    // - Retrieves the assigned command with the given key chord.
+    // - Can return nullopt to differentiate explicit unbinding vs lack of binding.
     // Arguments:
-    // - action: the shortcut action (an action type) we're looking for
+    // - keys: the key chord of the command to search for
     // Return Value:
-    // - the key chord that executes the given action
-    // - nullptr if the action is not bound to a key chord
-    Control::KeyChord ActionMap::GetKeyBindingForAction(const ShortcutAction& action) const
+    // - the command with the given key chord
+    // - nullptr if the key chord is explicitly unbound
+    // - nullopt if it is not bound
+    std::optional<Model::Command> ActionMap::_GetActionByKeyChordInternal(const Control::KeyChord& keys) const
     {
-        return GetKeyBindingForAction(action, nullptr);
+        if (const auto actionIDOptional = _GetActionIdByKeyChordInternal(keys))
+        {
+            if (!(*actionIDOptional).empty())

```

Comment:
```suggestion
            if (!actionIDOptional->empty())
```

i think optionals have `operator->` which lets you use `x->foo()` instead of the ugly `(*x).foo()`